plugins/storage/server/api/files.ts

```diff
@@ -49,17 +49,20 @@ router.get(
     const { key, sig } = ctx.input.query;
     const actor = ctx.state.auth.user;
     let attachment: Attachment | null;
+
     if (key) {
       attachment = await Attachment.findByKey(key);
-    } else {
-      attachment = await Attachment.findBySignature(sig!);
-    }
 
-    if (attachment.isPrivate) {
-      if (!actor) {
-        throw AuthenticationError();
+      if (attachment.isPrivate) {
+        if (!actor) {
+          throw AuthenticationError();
+        }
+        authorize(actor, "read", attachment);
       }
-      authorize(actor, "read", attachment);
+    } else if (sig) {
+      attachment = await Attachment.findBySignature(sig);
+    } else {
+      throw ValidationError("Must provide either key or signature");
```

Comment:
Well, this has already been handled in the `schema`.

Reply:
Yea, I know but I don't like using the `!` syntax – if we can make the zod type better that would work too